pallene/coder.lua

```diff
@@ -1683,16 +1716,56 @@ function Coder:generate_luaopen_function()
                     entry_point = self:lua_entry_point_name(upv.f_id),
                     ix = C.integer(self.upvalue_of_function[upv.f_id]),
                 }))
+            elseif tag == "coder.Upvalue.ImportedFunction" then
+                local imported_func = self.module.imported_functions[upv.f_id]
+                modules[imported_func.mod] = modules[imported_func.mod] or {}
+                imported_func.id = upv.f_id
+                imported_func._tag = "func"
```

Comment:
Assigning a `_tag` is strange. Should we create a proper tagged datatype, using `declare_type`?

Reply:
Fair enough, I think it would be better.